src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/TimeSeriesViewer.tsx

```diff
@@ -0,0 +1,308 @@
+        []
+    );
+    const handleOnRenderQuickTimeTitle = useCallback(
+        (options: IDropdownOption[]): JSX.Element => {
+            const option = options[0];
+            const iconStyles = { marginRight: '8px' };
+
+            return (
+                <div style={{ display: 'flex' }}>
+                    <Icon
+                        style={iconStyles}
+                        iconName="DateTime"
+                        aria-hidden="true"
+                    />
+                    <span>{option.text}</span>
+                </div>
+            );
+        },
+        [t]
+    );
+    const handleOnRenderYAxisTypeTitle = useCallback(
+        (options: IDropdownOption[]): JSX.Element => {
+            const option = options[0];
+            const iconStyles = { marginRight: '8px' };
+
+            return (
+                <div style={{ display: 'flex' }}>
+                    <Icon
+                        style={iconStyles}
+                        iconName="StackedLineChart"
+                        aria-hidden="true"
+                    />
+                    <span>{option.text}</span>
+                </div>
+            );
+        },
+        [t]
+    );
+    const handleOnRenderAggregationTypeTitle = useCallback(
+        (options: IDropdownOption[]): JSX.Element => {
+            const option = options[0];
+            const iconStyles = { marginRight: '8px' };
+
+            return (
+                <div style={{ display: 'flex' }}>
+                    <Icon
+                        style={iconStyles}
+                        iconName="AssessmentGroupTemplate"
+                        aria-hidden="true"
+                    />
+                    <span>{option.text}</span>
+                </div>
+            );
+        },
+        [t]
+    );
+    const updateXMinAndMax = useCallback(() => {
+        const nowInMillis = Date.now();
+        xMinDateInMillisRef.current =
+            nowInMillis - selectedQuickTimeSpanInMillis;
+        xMaxDateInMillisRef.current = nowInMillis;
+    }, [selectedQuickTimeSpanInMillis]);
+    const highChartSeriesData: Array<IHighChartSeriesData> = useMemo(
+        () =>
+            timeSeriesTwinList.length
+                ? transformADXTimeSeriesToHighChartsSeries(
+                      data,
+                      timeSeriesTwinList
+                  )
+                : [],
+        [data, timeSeriesTwinList]
+    );
+
+    // side effects
+    const prevQuery = usePrevious(query);
+    useEffect(() => {
+        if (query && query !== prevQuery) {
+            fetchTimeSeriesData();
+            updateXMinAndMax();
+        }
+    }, [query]);
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    return (
+        <div className={classNames.root}>
+            <Pivot
+                overflowBehavior={'menu'}
+                styles={classNames.subComponentStyles.pivot}
+            >
+                <PivotItem
```

Comment:
Currently I am working on the following PR for the Table and I am going to make it a separate internal component for the TimeSeriesViewer. I can make this other tab of "Chart" a separate component (and maybe even command bar as well) if you all agree. Since it was not too bad in the first pass I just laid out the components all in here. What do you think?

Reply:
Yeah. I am onboard with that. You mean something like having TimeSeriesViewerChart and TimeSeriesViewerTable?

Reply:
I like the sub component format since then you can have stories specific to each of those with more variants and whatnot. Seems like a nice encapsulation since there shouldn't need to be any communication between them right?

Reply:
I think each component probably only needs the series data as a prop. And that will help to keep the chart specific state and refs to itself.